README.md

```diff
@@ -1,30 +1,237 @@
 # Go Obscuro
-This is the reference implementation of the Obscuro Protocol.
-
-## Structure
-```
-|
-| - contracts: This folder contains the source code for the solidity Management contract, which will be deployed on Ethereum.  
-|               For now it's a very basic version that just stores rollups.
-| - go: The obscuro protocol code.
-|.   |
-|.   | - obscuronode: source code for the obscuro node. Note that the node is composed of two executables: "enclave" and "host".
-|.   |      | 
-|.   |      | - enclave: This is the component that is loaded up inside SGX.
-|.   |      | - host: The component that performs the networking duties.
-|.   |      | - obscuroclient: basic RPC library client code.
-|.   |      | - nodecommon:
-|
-| 
-| - integration: source code for end to end integration tests. 
-|
-
-```
 
-## Usage
-Todo
+This repository contains the reference implementation of the [Obscuro Protocol](https://whitepaper.obscu.ro/).
+
+*Note that this is still very much a work in progress, so there are many rough edges and unfinished components.*
 
 ## High level overview
+
+The typical blockchain node runs multiple services in a single process. For example:
+- P2P Service
+- RPC Service
+- Data storage
+- Transaction execution
+- Mempool
+- .. 
+
+Obscuro uses Trusted Execution Environments (TEE), like Intel SGX, to execute transactions in a confidential environment, which means we diverge from the typical architecture. 
+There are three main components of the architecture, each running as a separate process: the Enclave, the Host and the Wallet Extension.
+
+[//]: # (Todo - @joel - change the diagram to represent the Wallet Extension)
+![Architecture](design/obscuro_arch.jpeg)
+
+### I. The Enclave
+
+This is the core component of Obscuro which runs inside the TEE. 
+See [go/enclave](go/enclave)
+
+We use [EGo](https://www.edgeless.systems/products/ego/), an open source SDK for developing this confidential component.
+
+The Enclave exposes an [interface](go/common/enclave.go) over RPC which attempts to minimise the Trusted computing base(TCB).
```

Comment:
Should read Trusted "Computing Base (TCB)", or "trusted computing base (TCB)"?

Reply:
the latter. I'll add quotes